kombu/transport/redis.py

```diff
@@ -676,9 +676,8 @@ def _receive(self):
             ret.append(self._receive_one(c))
         except Empty:
             pass
-        if c.connection is not None:
-            while c.connection.can_read(timeout=0):
-                ret.append(self._receive_one(c))
+        while c.connection is not None and c.connection.can_read(timeout=0):
+            ret.append(self._receive_one(c))
         return any(ret)
 
     def _receive_one(self, c):
```

Comment:
In this function there's no chance that the connection we're using will change its value to None?